### PR TITLE
Fixes and improvements

### PR DESCRIPTION
**Pull request by nezorflame**
- fix warnings
- add additional Error
- update README
- bump Travis Go version

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ language: go
 go:
         - 1.7
         - 1.8
+        - 1.9
```
Comment (fiorix) [inline .travis.yml]: 1.10 was released few days ago, we should have that too

Comment (nezorflame) [reply .travis.yml]: Agree 👍

**Comment by nezorflame** [comment]
`go test -v -cover ./...` passes for me locally without any problems, while on Travis it takes forever.

**Comment by fiorix** [comment]
Wtf 1.10 -> 1.1 ? Maybe try "1.10" in quotes ?

**Comment by nezorflame** [comment]
@fiorix yep, funny bug 🤣 
Brackets fixed it.

**Comment by nezorflame** [comment]
Well, Travis fails on tests - it fails to move to the next directory and hungs.
Gonna need their support with this I think.

**Comment by fiorix** [comment]
Ok, I asked them: https://twitter.com/fiorix_/status/965957435189071872

**Comment by nezorflame** [comment]
I made an issue for that: https://github.com/travis-ci/travis-ci/issues/9247

**Comment by fiorix** [comment]
@nezorflame did you get an answer from them? I just restarted the tests now.

**Comment by nezorflame** [comment]
@fiorix nope. Seems that we would need their support again.

**Comment by nezorflame** [comment]
@fiorix I've identified the issue locally.
The problem is with the `TestConn` test in `smpp/conn_test.go`.
It hangs on the call:
```go
if _, err = c.Read(); err != nil {
		t.Fatal(err)
	}
```

**Comment by nezorflame** [comment]
Yup, fails on timeout:
```
goroutine 8 [IO wait]:
internal/poll.runtime_pollWait(0x15b8d20, 0x72, 0xc42005eaf0)
	/usr/local/opt/go/libexec/src/runtime/netpoll.go:173 +0x57
internal/poll.(*pollDesc).wait(0xc420120218, 0x72, 0xffffffffffffff00, 0x1333a60, 0x145a368)
	/usr/local/opt/go/libexec/src/internal/poll/fd_poll_runtime.go:85 +0x9b
internal/poll.(*pollDesc).waitRead(0xc420120218, 0xc420123000, 0x1000, 0x1000)
	/usr/local/opt/go/libexec/src/internal/poll/fd_poll_runtime.go:90 +0x3d
internal/poll.(*FD).Read(0xc420120200, 0xc420123000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/opt/go/libexec/src/internal/poll/fd_unix.go:157 +0x1dc
net.(*netFD).Read(0xc420120200, 0xc420123000, 0x1000, 0x1000, 0xc420132150, 0x10d0643, 0xc420130040)
	/usr/local/opt/go/libexec/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc42000e060, 0xc420123000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/opt/go/libexec/src/net/net.go:176 +0x6a
bufio.(*Reader).Read(0xc42007c240, 0xc4201400d0, 0x10, 0x10, 0x10, 0xc4201400d0, 0x0)
	/usr/local/opt/go/libexec/src/bufio/bufio.go:216 +0x238
io.ReadAtLeast(0x1332be0, 0xc42007c240, 0xc4201400d0, 0x10, 0x10, 0x10, 0x128c240, 0x1, 0xc4201400d0)
	/usr/local/opt/go/libexec/src/io/io.go:309 +0x86
io.ReadFull(0x1332be0, 0xc42007c240, 0xc4201400d0, 0x10, 0x10, 0x10, 0x12c88a0, 0xc420146240)
	/usr/local/opt/go/libexec/src/io/io.go:327 +0x58
github.com/fiorix/go-smpp/smpp/pdu.DecodeHeader(0x1332be0, 0xc42007c240, 0x1336680, 0xc42013e120, 0xc4201400c0)
	/Users/nezorflame/Go/src/github.com/fiorix/go-smpp/smpp/pdu/header.go:70 +0x97
github.com/fiorix/go-smpp/smpp/pdu.Decode(0x1332be0, 0xc42007c240, 0xc4200103a0, 0xc42004cf48, 0x18, 0xc4201321b0)
	/Users/nezorflame/Go/src/github.com/fiorix/go-smpp/smpp/pdu/codec.go:131 +0x50
github.com/fiorix/go-smpp/smpp/smpptest.(*conn).Read(0xc4201063e0, 0xc4201063e0, 0x0, 0x0, 0x0)
	/Users/nezorflame/Go/src/github.com/fiorix/go-smpp/smpp/smpptest/conn.go:50 +0x3b
github.com/fiorix/go-smpp/smpp/smpptest.(*Server).handle(0xc42007c1e0, 0xc4201063e0)
	/Users/nezorflame/Go/src/github.com/fiorix/go-smpp/smpp/smpptest/server.go:129 +0x190
created by github.com/fiorix/go-smpp/smpp/smpptest.(*Server).Serve
	/Users/nezorflame/Go/src/github.com/fiorix/go-smpp/smpp/smpptest/server.go:108 +0x45
```

**Comment by nezorflame** [comment]
Fixed my derp, somewhere in the process of warning fixes I've lost a whole function call, lol.

**Comment by nezorflame** [comment]
@fiorix should be good to go now.

**Comment by fiorix** [comment]
omg

**Pull request merged**